Proposal writeup and Function for numerical plots (#12)
UBC-MDS/simpler_eda: README.md

## Overview

Exploratory Data analysis (EDA) is an important step in any data analysis.  However, carrying out EDA with the Altair package requires a lot of coding effort. Moreover, it assumes a basic knowledge of functions and grammar of graphics syntax that are appropriate for visualizing categorical and numerical variables. The simpler_eda package addresses this issue by providing functions that are tailored to produce categorical, numerical and correlation plots using a single line of code. Furthermore, the package provides customization capability for the plots based on specific user needs (theme, title, font, size and etc.). The users are able to spend more time on analyzing the data set and less time configuring Altair plot settings.

## Installation

```bash
$ pip install -i https://test.pypi.org/simple/ simpler_eda
```

## Functions

This package contains three functions, each that accepts a pandas `DataFrame` for EDA. No specific data set is required for the functions to work. Each functions will have it's own required and optional arguements to configure the properties of the plot. 

> 

Reviewer: update description according to docstring (after updated PR) pls

Author: updated

> 1. `correlation_eda_plot:`  Plot a correlation map with the given dataframe object and a list of numerical features. Users are allow to set multiple arguments regarding the setting of the correlation plot from color schemes, plot width, height and plot title.

Reviewer: Users are **_allowed_** to set multiple arguments regarding the setting of the correlation plot **_including_** color schemes, plot width, height and plot title.

Author: Good observation!

2. `numerical_eda_plot:` This function takes in a data frame object, two numeric columns, and produces either a scatter or line plot to visualize the relationship of the two features. User can optionally change default arguements for plot-type, color/fill, title, size of text, color-scheme and apply logarithm scaling on the x and y axis.

3. `categorical_eda_plot:` This function takes in a data frame object, a numerical and a categorical column and produces either a jitter, violin, barchar and boxplot (two categoricals) for comparison. User can optionally change default arguements for plot-type, color/fill, title, size of text and color-scheme.

## How the simpler_eda package fits into the Python ecosystem

> The simpler_eda package improves upon existing functions in the Altair packages library. Altair already includes many useful functions to visualize relationship between numerical and categorical features with the use of grammar of graphics syntax. But often this is quite cumbersome and prone to errors, simpler_eda package provides convenience by allowing user to perform EDA with a single line of code. There are a number of packages that already provides similar functionality in the Python Ecosystem, but most of them are not easily customizable. Some examples of the packages are [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews).

Reviewer 2: ```suggestion
The simpler_eda package improves upon existing functions in the Altair packages library. Altair already includes many useful functions to visualize the relationship between numerical and categorical features with the use of grammar of graphics syntax. But often this is quite cumbersome and prone to errors, simpler_eda package provides convenience by allowing users to perform EDA with a single line of code. There are a number of packages that already provide similar functionality in the Python Ecosystem, such as [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews). However, most of them are not easily customizable. Our simpler_eda package allows flexibility from plot types to plot titles.
```

> The simpler_eda package improves upon existing functions in the Altair packages library. Altair already includes many useful functions to visualize relationship between numerical and categorical features with the use of grammar of graphics syntax. But often this is quite cumbersome and prone to errors, simpler_eda package provides convenience by allowing user to perform EDA with a single line of code. There are a number of packages that already provides similar functionality in the Python Ecosystem, but most of them are not easily customizable. Some examples of the packages are [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews).

Reviewer: The simpler_eda package improves upon existing functions in the Altair packages library. There are a number of packages that already provides similar functionality in the Python Ecosystem, but most of them are not easily customizable. Some examples of the packages are [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews).

Author: Used Chuck's paragraph for this description.

## Functions

This package contains three functions, each that accepts a pandas `DataFrame` for EDA. The EDA functions can be used with a dataset with numerical and categorical features. Each functions will have it's own required and optional arguments to configure the properties of the plot. 

> 1. `correlation_eda_plot:`  Plot a correlation map with the given dataframe object and a list of numerical features. Users are allow to set multiple arguments regarding the setting of the correlation plot from color schemes, plot width, height and plot title.

Reviewer 2: ```suggestion
1. `correlation_eda_plot:`  Plot a correlation map with the given dataframe object and a list of numerical features. Users are allowed to set multiple arguments regarding the setting of the correlation plot from color schemes, plot width, height, and plot title.
```

> 2. `numerical_eda_plot:` This function takes in a data frame object, two numeric columns, and produces either a scatter or line plot to visualize the relationship between the two features. User can optionally change default arguments for plot-type, color, title, size of text, color-scheme, and toggle log transformation for the x and y axis.

Reviewer 2: ```suggestion
2. `numerical_eda_plot:` This function takes in a data frame object, two numeric columns, and produces either a scatter or line plot to visualize the relationship between the two numerical features. Users can optionally change default arguments for plot-type, color, title, size of text, color-scheme, and toggle log transformation for the x and y axis.
```

> 3. `categorical_eda_plot:`  This function takes in a data frame object and one categorical feature, to produce a histogram plot that visualizes the distribution of the feature. User can also choose to plot density graph of the feature by specifing in plot_type. The function also offers customization on color, plot title, font size, color-scheme, plot size, opacity level and facet factor.

Reviewer 2: ```suggestion
3. `categorical_eda_plot:`  This function takes in a data frame object and one categorical feature, to produce a histogram plot that visualizes the distribution of the feature. Users can also choose to plot the density graph of the feature by specifying in plot_type. The function also offers customization on color, plot title, font size, color-scheme, plot size, opacity level, and facet factor.
```

## How the simpler_eda package fits into the Python ecosystem

> The simpler_eda package improves upon existing functions in the Altair packages library. Altair already includes many useful functions to visualize the relationship between numerical and categorical features with the use of grammar of graphics syntax. But often this is quite cumbersome and prone to errors, simpler_eda package provides convenience by allowing users to perform EDA with a single line of code. There are a number of packages that already provide similar functionality in the Python Ecosystem, such as [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews). However, most of them are not easily customizable. Our simpler_eda package allows flexibility from plot types to plot titles.

Reviewer 2: ```suggestion
The simpler_eda package improves upon existing functions in the Altair packages library. Altair already includes many useful functions to visualize the relationship between numerical and categorical features with the use of grammar of graphics syntax. But often this is quite cumbersome and prone to errors, simpler_eda package provides convenience by allowing users to perform EDA with a single line of code. There are a number of packages that already provide similar functionality in the Python Ecosystem, such as [pandas profiling](https://github.com/pandas-profiling/pandas-profiling), [DataPrep](https://docs.dataprep.ai/index.html), [SweetViz](https://pypi.org/project/sweetviz/) and [Holoviwes](https://github.com/holoviz/holoviews). However, most of them are not easily customizable. Our simpler_eda package allows flexibility from plot types, color scheme, to plot titles.
```
